src/auth.nim

```diff
@@ -197,13 +198,55 @@ proc initAccountPool*(cfg: Config; path: string) =
   elif fileExists(path):
     log "Parsing JSON guest accounts file: ", path
     accountPool = parseGuestAccounts(path)
-  else:
-    echo "[accounts] ERROR: ", path, " not found. This file is required to authenticate API requests."
+  elif not cfg.guestAccountsUsePool:
+    echo "[accounts] ERROR: ", path, " not found. This file is required to authenticate API requests. Alternatively, configure the guest account pool in nitter.conf"
     quit 1
 
-  let accountsPrePurge = accountPool.len
   accountPool.keepItIf(not it.hasExpired)
 
   log "Successfully added ", accountPool.len, " valid accounts."
-  if accountsPrePurge > accountPool.len:
-    log "Purged ", accountsPrePurge - accountPool.len, " expired accounts."
+
+proc updateAccountPool*(cfg: Config) {.async.} =
+  if not cfg.guestAccountsUsePool:
+    return
+
+  # wait for a few seconds before fetching guest accounts, so that
+  # /.well-known/... is served correctly
+  await sleepAsync(10 * 1000)
+
+  while true:
+    if accountPool.len == 0:
+      log "fetching more accounts from service"
+
+      let client = newAsyncHttpClient()
+
+      try:
+          let resp = await client.get($(cfg.guestAccountsPoolUrl ? {"id": cfg.guestAccountsPoolId, "auth": cfg.guestAccountsPoolAuth}))
+          let guestAccounts = await resp.body
+
+          log "status code from service: ", resp.status
+
+          for line in guestAccounts.splitLines:
+            if line != "":
+              accountPool.add parseGuestAccount(line)
```

Comment:
Minor change here for consistency + indent fix:
```suggestion
      let client = newAsyncHttpClient()

      try:
        let 
          resp = await client.get(cfg.guestAccountsPoolUrl ? {"id": cfg.guestAccountsPoolId, "auth": cfg.guestAccountsPoolAuth})
          guestAccounts = await resp.body

        log "status code from service: ", resp.status

        for line in guestAccounts.splitLines:
          if line.len == 0:
            accountPool.add parseGuestAccount(line)
```

Reply:
oops!

Reply:
Also, you can add a user-agent if you want, just `newAsyncHttpClient("...")`